lite/tests/classification/test_dataloader.py

```diff
@@ -0,0 +1,44 @@
+import json
+
+import pytest
+from valor_lite.classification import Classification, DataLoader
+
+
+def test_no_data():
+    loader = DataLoader()
+    with pytest.raises(ValueError):
+        loader.finalize()
+
+
+def test_valor_integration():
+
+    gt_json = '{"datum": {"uid": "2hVjLfwXSkbWmFhaISWOT2mHTbx", "metadata": {"width": {"type": "integer", "value": 224}, "height": {"type": "integer", "value": 224}}}, "annotations": [{"task_type": "classification", "metadata": {}, "labels": [{"key": "class_label", "value": "RED TAILED HAWK", "score": null}], "bounding_box": null, "polygon": null, "raster": null, "embedding": null}]}'
+    pd_json = '{"datum": {"uid": "2hVjLfwXSkbWmFhaISWOT2mHTbx", "metadata": {"width": {"type": "integer", "value": 224}, "height": {"type": "integer", "value": 224}}}, "annotations": [{"task_type": "classification", "metadata": {}, "labels": [{"key": "class_label", "value": "AMERICAN AVOCET", "score": 1.7822019060531602e-07}, {"key": "class_label", "value": "AVADAVAT", "score": 0.0005651581450365484}, {"key": "class_label", "value": "BLACK TAIL CRAKE", "score": 0.00014690875832457095}, {"key": "class_label", "value": "BLUE MALKOHA", "score": 0.07310464978218079}, {"key": "class_label", "value": "CAMPO FLICKER", "score": 0.9117385149002075}, {"key": "class_label", "value": "CHUKAR PARTRIDGE", "score": 0.0014892773469910026}, {"key": "class_label", "value": "CRESTED KINGFISHER", "score": 0.012080634944140911}, {"key": "class_label", "value": "DOUBLE BRESTED CORMARANT", "score": 0.0004424828803166747}, {"key": "class_label", "value": "EMU", "score": 2.4317616407643072e-05}, {"key": "class_label", "value": "GREAT KISKADEE", "score": 0.00040792522486299276}], "bounding_box": null, "polygon": null, "raster": null, "embedding": null}]}'
+
+    gt = json.loads(gt_json)
+    pd = json.loads(pd_json)
+
+    loader = DataLoader()
+    loader.add_data_from_valor_dict([(gt, pd)])
+    loader.finalize()
+
+    assert loader._evaluator._detailed_pairs.shape == (10, 5)
+
+    assert set(loader._evaluator.label_key_to_index.keys()) == {
+        "class_label",
+    }
+    assert len(loader._evaluator.index_to_label) == 11
+    assert loader._evaluator.n_datums == 1
+
+
+def test_mismatch_label_keys(
+    classifications_no_groundtruths: list[Classification],
+    classifications_no_predictions: list[Classification],
+):
+    loader = DataLoader()
+
+    with pytest.raises(ValueError):
+        loader.add_data(classifications_no_groundtruths)
+
+    with pytest.raises(ValueError):
+        loader.add_data(classifications_no_predictions)
```

Comment:
in `test_evaluate_classification_model_with_no_predictions`, we show that we want users to be able to pass an empty set of predictions and still get back classification metrics (with zeroes for all values). doesn't seem like this test was added, and this behavior may be disallowed as shown by this line